### PR TITLE
Add Eigen3

### PR DESCRIPTION
**Pull request by KorigamiK**
Eigen3 is an awesome C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms.

It is fully inline so there should not be any static or shared libraries required to link.

I would love to know how you would go about linking this one in a cmakelists.txt :D

### CODE REVIEW

**Comment by KorigamiK** [comment]
I've created a [minimal working example](https://gist.github.com/KorigamiK/97fb9536ce9ae29aff29f7d719483f29) to test that it works :)

**Comment by sharkwouter** [comment]
Thanks! I'll give this a test and do a review.

**Review by sharkwouter — CHANGES_REQUESTED** [review]
At the moment I can't accept this yet, because the generated package is empty. I made 2 suggestions which should resolve that.

You might also want to make sure the pkgconfig file revers to $PSPDEV in a similar fashion the SDL packages does it: https://github.com/pspdev/psp-packages/blob/9e1eb362ac8ef95688349c23f11baa0f8f8acc05/sdl2/PSPBUILD#L17

Other than that this looks good. Thanks for the PR, it's good to see other people make some.

File: eigen3/PSPBUILD [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+pkgdesc="C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms"
+arch=('mips')
+sha256sums=('8586084f71f9bde545ee7fa6d00288b264a2b7ac3607b974e54d13e7162c1c72')
+
+
+build() {
+  cd eigen-$pkgver
+  mkdir build
+  cd build
+  cmake .. -DCMAKE_TOOLCHAIN_FILE="$PSPDEV/psp/share/pspdev.cmake" -DCMAKE_INSTALL_PREFIX="$PSPDEV/psp" -DCMAKE_BUILD_TYPE=Release -Wno-dev -DBUILD_SHARED_LIBS=OFF
```
[the remark this answers is not in the document]
Comment (KorigamiK) [reply eigen3/PSPBUILD:18]: Aah, if I understand correctly, the `/psp` directory will automatically be installed to `$PSPDEV/psp`. Is that right?

As of now I can see that the Library is actually installed and the files are saved in the correct location.

[the remark this answers is not in the document]
Comment (sharkwouter) [reply eigen3/PSPBUILD:18]: Exactly :D

**Comment by KorigamiK** [comment]
Thanks for the changes, I can't test them at the moment but please change them as you desire.

**Pull request merged**